build.gradle

```diff
@@ -154,6 +154,9 @@ dependencies {
   implementation group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.11.0'
   implementation group: 'com.google.guava', name: 'guava', version: '29.0-jre'
   implementation group: 'commons-io', name: 'commons-io', version: '2.7'
+  implementation group: 'uk.gov.hmcts.reform', name: 'service-auth-provider-client', version: '3.1.1'
+  implementation group: 'uk.gov.hmcts.reform', name: 'idam-client', version: '1.4-beta'
+  implementation group: 'uk.gov.hmcts.reform', name: 'core-case-data-store-client', version: '4.7.0'
```

Comment:
So there will bring whole lot of spring dependencies and giving they are clients - feign dependencies as well :see_no_evil: . this repo has removed all spring relations as it doesn't need to run any sort of app nor magical injection procedures. and it's not an app to run any autoconfig

How about we raise a PR per client whichever you need but instead of sticking in all those ^ just re-use `RestAssured` functionality. I'm pretty sure when I did a spike on idam oauth2 stuff I have commented detailed template of `RestAssured` call and successfully retrieved needed data. Same can be done for all of them.

Before making this comment I scrolled a bit forwards and backwards in this PR and it is sort of confusion - all of a sudden spring's stereotypes come in and that would mean all that magical AOP autoconfig will have to go in as well for fun tests to run what they need to run. Is this what we want?